### ISSUE TITLE
Required  type of transportation at `routes` level may lack flexibility (multi-modal routes)

### ISSUE THREAD

**Issue by barbagus** [google/transit#409]
The French regional railways [TER](https://en.wikipedia.org/wiki/Transport_express_r%C3%A9gional) operates both trains and buses on regular routes all over the country. They do publish a GTFS dataset daily: [https://eu.ftp.opendatasoft.com/sncf/gtfs/export-ter-gtfs-last.zip](https://eu.ftp.opendatasoft.com/sncf/gtfs/export-ter-gtfs-last.zip)

Some routes are designed to be operated with both trains and buses depending on the trip. You can download a human readable (PDF) schedule for such a route [here](https://ter-fiches-horaires.sncf.fr/publish/A19%20et%20cars%20Colmar%20Metzeral%20%28du%2021%20ao%C3%83%C2%83%C3%82%C2%BBt%20au%209%20d%C3%83%C2%83%C3%82%C2%A9cembre%202023%29%20MAJ%201008.pdf) (fr).
On the schedule table of the first page, the `12:44`, `20:28` and `22:00` trips are clearly identified with a blue bus icon (![image](https://github.com/google/transit/assets/63465740/622e59fd-f76e-48f8-bbae-9e0ce588b7bf)) which is defined in the legend as regular trip carried out with bus. I insist, those are not exceptions (that could be handled through GTFS-RT).

In the current GTFS specification, the "type of transportation" is a `routes` attribute and as such, cannot *elegantly* handle such use case.

## Workarounds
There are few ways around this (that I can think about) but none of them are without drawback (technically or semantically).

1. **Route duplication**: create one route regrouping the "train trips" and one route the "bus trips". Somehow it is kind of what they are doing in the PDF version of their schedules. If you scroll down to the last pages you will notice that after the general "Route 19" schedule there is a specific section for a "Route 119" that only shows bus trips.
They do that to precisely explain to the user where the bus stops are relative to the train station. Usually very close (few meters) but sometimes a bit further.
However they still chose to include the bus trips in the general schedule of "Route 19" because semantically **it is the same route** and people might not bother scrolling down few pages to find the additional trips.

2. **Loose use of `routes.route_type` (breaking the specs)**: that's actually what they implement in their GTFS dataset. That particular route is marked with a `2 / Rail` value (because it is mostly carried out by trains I suppose) but they do encode the actual type of transportation within their `stops.ID` pattern. For instance, at the same station, they'd have a `StopPoint:OCETrain TER-87182394` stop and a `StopPoint:OCECar TER-87182394` stop. Then, as a GTFS consumer, one can infer the type of transportation by checking that all stops along the trip are `*OCETrain*` or `*OCECar*` (*car* means bus in French).
_Note:_ Although technically, having to read the whole stops sequence to infer transportation type is not ideal, I find it very interesting and somehow legitimate for the type of transportation to be a `stops | location_type=0` attribute actually: after all, "train stops" and "bus stops" are very commonly distinguishable entities from the *real world*.

## Suggestions for modification of the specs
* Implement a way to *relate* routes together, so in this example the "route duplication" workaround could be addressed properly.
* Implement type of transportation overriding scheme at `trips` level (see #358) or at `stops` level as in the second workaround. Such a overriding would mean to specify the `routes.route_type` to be a **default value**  and/or to add a "Mixed" value to the enum and handle the necessary conditional requirements.

**Comment by barbagus** [comment]
My bad, I copied my post over there. We can close this issue I suppose.

**Comment by barbagus** [comment]
The call to modification I posted doubles with #358, but I guess we could also co sider ways to regularize le observes workarounds I mentioned.

1. Implement a route grouping mechanism to regularize the "route duplication"
2. Make `routes.route_type` conditionally optional and "move" the transportation type to `trips` as suggested in #358 or down to `stops` as observed in the second workaround.

**Comment by skinkie** [comment]
Can you board and alight at the same stops?

**Comment by barbagus** [comment]
No. Usually the same parent station but, as I mentioned, they do have separate stops, one for trains, one for buses.

**Comment by skinkie** [comment]
Then I think it is plainly wrong to use the same _route_. I would also argue that in every other standard TER uses they have to split them in two separate lines, which may for the public share the same public code, but never is this thing combined into one timetable.

**Comment by barbagus** [comment]
I agree that the "route duplication" is my favourite workaround.
However whether it is wrong for them to do so, it seems that they actually do combine... Upon talking with the administrators of that datasets, they confirmed that the gtfs generation is the end of a very long pipeline using their own information system as a source.
And they do show them combined in one time table (see the linked pdf exemple)

One thing is sure, we might not succeed to have *them* change their IS... the question is then, as a community standard... should gtfs support this ?

**Comment by skinkie** [comment]
Can you show me their NeTEx output for this line?

**Comment by eliasmbd** [comment]
Thank you for bringing this issue to the community's attention. Your contribution is clear and well structured! 🙏

> One thing is sure, we might not succeed to have _them_ change their IS... the question is then, as a community standard... should gtfs support this ?
You also ask an essential question here especially for an edge case like this. Have you been in communication with the TER authorities to figure out their thinking behind this case?

**Comment by barbagus** [comment]
> Have you asked the TER authorities about their thinking behind this case?

Not directly, I did approach them to get confirmation of the fact that their `stops.stop_id` do follow the pattern I mention in the original issue, i.e. `StopPoint:OCECar*` means bus and `StopPoint:OCETrain` means train (they also have a prefix for Tram..).
Also, I invited them to checkout this issue post.

**Comment by miklcct** [comment]
Shall we close this and merge with #358 ?

**Comment by barbagus** [comment]
Looking again at #358, it is still about temporary replacements which is related to this but not the same thing. Despite the lack of traction for this issue, I believe it still stands.

**Comment by felixguendling** [comment]
I would hope that if we find a modelling to enable this use case here, it also solves the #358 use case and vice versa.
